base/io.jl

```diff
@@ -167,24 +167,50 @@ The text is assumed to be encoded in UTF-8.
 readuntil(filename::AbstractString, args...) = open(io->readuntil(io, args...), filename)
 
 """
-    readline(stream::IO=STDIN)
-    readline(filename::AbstractString)
+    readline(stream::IO=STDIN, chomp::Bool=false)
+    readline(filename::AbstractString, chomp::Bool=false)
+
+Read a single line of text from the given I/O stream or file (defaults to `STDIN`).
+Lines in the input can end in `'\\n'` or `'\\r\\n'`. When reading from a file, the text is
+assumed to be encoded in UTF-8. If `chomp=false` trailing newline character(s) will be included
+in the output (if reached before the end of the input); otherwise newline characters(s)
+are stripped from result.
+"""
+function readline(filename::AbstractString, chomp = false)
+    open(filename) do f
+          readline(f, chomp)
+    end
+end
 
-Read a single line of text, including a trailing newline character (if one is reached before
-the end of the input), from the given I/O stream or file (defaults to `STDIN`).
-When reading from a file, the text is assumed to be encoded in UTF-8.
-"""
-readline(filename::AbstractString) = open(readline, filename)
+function readline(s::IO=STDIN, chomp::Bool=false)
+    chomp || return String(readuntil(s, 0x0a))
 
-"""
-    readlines(stream::IO)
-    readlines(filename::AbstractString)
+    line = readuntil(s, 0x0a)
+    i = length(line)
+    if i < 1 || line[i] != 0x0a
```

Comment:
No idea, would need to check. You're right that it's probably going to be negligible except maybe for very short strings, so maybe it's OK to have the first branch precisely for them.